oscarlab/graphene
# Loader rework pre-cleanup

## Description of the changes 

This PR contains a bunch of simple clean-ups which were required by the more complex clean-ups (coming soon), which are needed by the loader rework (coming less soon).

## How to test this PR? 

CI + check compilation warnings on GCC before and now (see commit message about the `-Wno-gnu-alignof-expression` fix).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/1895)
